### PR TITLE
Task configure sa: 'Add image pull secrets to a service account' sect…

### PR DESCRIPTION
**Pull request by scoulomb**
Improvement proposal to show:
- editing yaml file is an alternative of path
- `spec.imagePullSecrets` field added if in namespace but also using correct sa  (line 264)

### CODE REVIEW

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 1]
```diff
@@ -183,27 +183,38 @@ The content of `token` is elided here.
...
 ## Add ImagePullSecrets to a service account
 
-First, create an imagePullSecret, as described [here](/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod).
-Next, verify it has been created. For example:
+### Create an imagePullSecret
 
-```shell
-kubectl get secrets myregistrykey
-```
+- Create an imagePullSecret, as described [here](/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod).
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:188]: nit:
```suggestion
- Create an imagePullSecret, as described in [Specifying ImagePullSecrets on a Pod](/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod).
```

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 2]
```diff
@@ -183,27 +183,38 @@ The content of `token` is elided here.
...
+            --docker-username=DUMMY_USERNAME --docker-password=DUMMY_DOCKER_PASSWORD \
+            --docker-email=DUMMY_DOCKER_EMAIL
+    ```
 
-```
-NAME             TYPE                              DATA    AGE
-myregistrykey    kubernetes.io/.dockerconfigjson   1       1d
-```
+- Verify it has been created.
+   ```shell script
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:197]: nit: `shell script` should be just `shell`

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 3]
```diff
@@ -183,27 +183,38 @@ The content of `token` is elided here.
...
 
 Next, modify the default service account for the namespace to use this secret as an imagePullSecret.
 
+
 ```shell
 kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "myregistrykey"}]}'
 ```
 
-Interactive version requires manual edit:
+Alternatively we can use `kubectl edit` or edit manually yaml files as shown below:
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:217]: This wording would be more idiomatic:
```suggestion
You can instead use `kubectl edit`, or manually edit the YAML manifests as shown below:
```

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 4]
```diff
@@ -248,10 +259,18 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
 kubectl replace serviceaccount default -f ./sa.yaml
 ```
 
-Now, any new pods created in the current namespace will have this added to their spec:
+### Verify imagePullSecrets was added to pod spec
+
+Now, any new pods created in the current namespace and using default service account will have `spec.imagePullSecrets` field added:
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:264]: How about this wording:
```suggestion
Now, when a new Pod is created in the current namespace and using the default ServiceAccount, the new Pod has its  `spec.imagePullSecrets` field set automatically:
```

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 5]
```diff
@@ -248,10 +259,18 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
...
+
+````shell
+kubectl run nginx --image=nginx --restart=Never
+kubectl get pod nginx -o yaml | grep -A 1 imagePullSecrets
+````
+
+Output is
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:271]: nit:
```suggestion
The output is:
```

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 6]
```diff
@@ -248,10 +259,18 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
...
 ```
 
-Now, any new pods created in the current namespace will have this added to their spec:
+### Verify imagePullSecrets was added to pod spec
+
+Now, any new pods created in the current namespace and using default service account will have `spec.imagePullSecrets` field added:
+
+````shell
+kubectl run nginx --image=nginx --restart=Never
+kubectl get pod nginx -o yaml | grep -A 1 imagePullSecrets
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:268]: This could also use a [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) query. How about taking that option (rather than `grep`)?

Comment (scoulomb) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: Yes, it is much better!

**Comment by scoulomb** [comment]
> @scoulomb I've added some feedback. Anything marked as a “nit” means that I think this is a good idea to fix, but also that this is fine to merge without addressing. Also, the questions genuinely are questions rather than mandates.
> 
> I hope you find these suggestions useful. If you would like to ask questions about this feedback please feel free to do that.

@sftim Thanks a lot for all your suggestions! it was very useful. I updated.

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 7]
```diff
@@ -248,12 +259,19 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
...
+Now, when a new Pod is created in the current namespace and using the default ServiceAccount, the new Pod has its  `spec.imagePullSecrets` field set automatically:
+
+````shell
+kubectl run nginx --image=nginx --restart=Never
+kubectl get pod nginx -o=jsonpath='{.spec.imagePullSecrets[0].name}'
+````
+
+The output is:
+
+```shell
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:273]: nit: this is sample output, not shell code

**Comment by scoulomb** [comment]
> There's a snag but it's not a big one. @scoulomb if you wanted to fix that please do feel free to send in a new PR.
> 
> /lgtm
> /approve

@sftim thanks I will fix this
